core/src/main/java/io/questdb/cairo/TxReader.java

```diff
@@ -53,7 +54,17 @@
     protected int attachedPartitionsSize = 0;
     private MemoryMR roTxMem;
 
-    public TxReader(FilesFacade ff, @Transient Path path, int partitionBy) {
+    public TxReader(FilesFacade ff) {
+        this.ff = ff;
+    }
+
+    @Override
+    public void clear() {
+        close();
+    }
+
+    public TxReader ofRO(@Transient Path path, int partitionBy) {
```

Comment:
it would be good if this method would restore path to its original length, it is not super clear from the API if path mutates or not

Reply:
done in https://github.com/questdb/questdb/pull/1778/commits/0e9e766e63f19b305aafbcd5d6f7e8c9276ecde9